new_webapp/src/reducers/candidateReducer.js

```diff
@@ -68,16 +78,36 @@ export function candidateReducer(state = initialCandidateState, action){
                 isFetching: action.payload.status === LOADING,
             };
         case Consts.ADD_CANDIDATE:
+            const candidates = action.payload.candidates;
+            const newResults = candidates.reduce((accumResults, candidate) => {
+                const currentPrograms = accumResults.map((existCandidate) => existCandidate.code);
+                const idx = currentPrograms.indexOf(candidate.code);
+                if (idx !== -1)  { // this is an old candidate, but new examples come
+                    const newExamples = candidate.examples.map(toExample);
+                    const targetExample = accumResults[idx].examples;
+                    accumResults[idx].examples = targetExample.concat(newExamples);
```

Comment:
nit: Use an early return here, it makes code easier to read, since you can just stop reading when you the `return` and start thinking about the next case.